contracts/token/ERC20/extensions/draft-ERC20Bridgeable.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.27;
+
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {IERC7802} from "../../../crosschain/interfaces/draft-IERC7802.sol";
+import {ERC165, IERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+
+/**
+ * @dev ERC20 extension that implements the standard token interface according to
+ * https://github.com/ethereum/ERCs/blob/bcea9feb6c3f3ded391e33690056635d722b101e/ERCS/erc-7802.md[ERC-7802].
+ *
+ * NOTE: To implement a crosschain gateway for a chain, consider using an implementation if {IERC7786} token
+ * bridge (e.g. {AxelarGatewaySource}, {AxelarGatewayDestination}).
+ */
+abstract contract ERC20Bridgeable is ERC165, ERC20, IERC7802 {
+    /// @dev Modifier to restrict access to the token bridge.
+    modifier onlyTokenBridge() {
+        _checkTokenBridge(msg.sender);
+        _;
+    }
+
+    /**
+     * @dev Checks if the caller is a trusted token bridge. MUST revert otherwise.
+     *
+     * Developers should implement this function using an access control mechanism that allows
+     * customizing the list of allowed senders. Consider using {Ownable}, {AccessControl} or {AccessManager}.
+     */
+    function _checkTokenBridge(address caller) internal virtual;
```

Comment:
usually we put these virtual not implemented function at the end of the contract.